diff --git shadertoys/raymarching_-_primitives/image.frag shadertoys/raymarching_-_primitives/image.frag
index 2a48cd4..003e29d 100644
--- shadertoys/raymarching_-_primitives/image.frag
+++ shadertoys/raymarching_-_primitives/image.frag
@@ -272,7 +272,7 @@ float sdPyramid( in vec3 p, in float h )
     
     // symmetry
     p.xz = abs(p.xz);
-    p.xz = (p.z>p.x) ? p.zx : p.xz;
+    p.xz = (p.z>p.x) ? (vec2)p.zx : p.xz;
     p.xz -= 0.5;
 	
     // project into face plane (2D)
